draft-ietf-quic-transport.md

```diff
@@ -240,23 +240,58 @@ Application:
 
 ## Notational Conventions
 
-Packet and frame diagrams in this document use the format described in Section
-3.1 of {{?RFC2360}}, with the following additional conventions:
+Packet and frame diagrams in this document use a bespoke format. The purpose of
+this format is to summarize, not define, protocol elements. Prose defines the
+complete semantics and details of structures.
 
-\[x\]:
-: Indicates that x is optional
+Complex fields are named and then followed by a list of fields surrounded by a
+pair of matching braces. Each field in this list is separated by commas.
+
+Individual fields include length information, plus indications about fixed
+value, optionality, or repetitions. Individual fields use the following
+notational conventions:
 
 x (A):
 : Indicates that x is A bits long
 
-x (A/B/C) ...:
-: Indicates that x is one of A, B, or C bits long
-
-x (i) ...:
+x (i):
 : Indicates that x uses the variable-length encoding in {{integer-encoding}}
 
-x (*) ...:
-: Indicates that x is variable-length
+x (A..B):
+: Indicates that x can be any length from A to B; A can be omitted to indicate
+  a mimumum of zero bits and B can be omitted to indicate no set upper limit;
+  values always end on an octet boundary
+
+x (?) = C:
+: Indicates that x has a fixed value of C
+
+x (?) = C..D:
+: Indicates that x has a value in the range from C to D, inclusive
+
+\[x (E)\]:
+: Indicates that x is optional (and has length of E)
+
+x (E) ...:
+: Indicates that x is repeated zero or more times (and that each instance is
+  length E)
+
+By convention, individual fields reference a complex field by using the name of
+that the complex field.
+
+For example:
+
+~~~
+Example Structure {
+  One-bit Field (1),
+  7-bit Field with Fixed Value (7) = 61,
+  Variable-Length Field (8..24),
+  Field With Minimum Length (16..),
```

Comment:
```suggestion
  Arbitrary-Length Field (..),
  Variable-Length Field (8..24),
  Field With Minimum Length (16..),
```

Omitting both ends is used in the document and consistent with the definition, but not clearly spelled out or used in an example.